### ISSUE TITLE
Consume input from sys.stdin

### ISSUE THREAD

**Issue by pdawyndt** [dodona-edu/papyros#221]
I would have a assumed that batch input could also be consumed this way

```python
import sys

for name in sys.stdin:
    print(f'Hello, {name}')
```

but apparently the batch input isn't bound deep enough for the `sys.stdin` iterator to be able to consume it.

![image](https://user-images.githubusercontent.com/5736113/171338864-87363e17-1425-4f86-9760-154e4a7c4631.png)

**Comment by winniederidder** [comment]
We override the builtin function `input` and the method `readline` of sys.stdin. Apparently, its iterator does not use readline. I must say that I have also never tried to use sys.stdin as an iterable.

**Comment by pdawyndt** [comment]
Isn’t it the other way round? Readline using the iterator protocol to fetch the next line from stdin?

**Comment by pdawyndt** [comment]
Readline using sys.stdin.next?

**Comment by winniederidder** [comment]
> Readline using sys.stdin.next?

Well, readline has some extra options. It might be the case that it internally calls next, so then we could override that instead of readline. I'm currently doing some experimenting, but sadly the internals of sys.stdin are implemented in https://github.com/python/cpython/blob/main/Python/sysmodule.c, making it not quite as straightforward to figure out what happens precisely.

**Comment by winniederidder** [comment]
Overriding sys.stdin.__next__ is easy and works. sys.stdin.__iter__ / iter(sys.stdin) simply return sys.stdin, so that is also no issue. However, next(sys.stdin) doesn't seem to call sys.stdin.__next__, it seems to call something else, resulting in an immediate StopIteration. Perhaps @alexmojaki can shine some light on this?

**Comment by alexmojaki** [comment]
Released a fix in python_runner 0.6.1

**Comment by winniederidder** [comment]
Integrated by re-deploying, so this issue has been resolved. Thanks @alexmojaki

**Comment by pdawyndt** [comment]
I would suggest to raise a StopIteration when reaching the end of batch input.

**Comment by winniederidder** [comment]
We chose to keep the batch input editable at runtime. Therefore, it would make no sense to raise StopIteration when reaching the "end", as there actually is no end. The actual behaviour of sys.stdin as an iterable is also to keep going. Users using this method always need some sort of while-loop to break out of this themselves

**Comment by pdawyndt** [comment]
I would think of it as redirecting batch input as a file to stdin, as an alternative to the interactive input. If we keep it this way, at least there should be an explicit EOF option when providing interactive input.

**Comment by winniederidder** [comment]
When trying this locally in a terminal, Ctrl-D does not work while iterating over sys.stdin, so this would seemingly deviate from the standard. As mentioned earlier, exercises in Dodona will usually ask the user to use a while-loop that tests for something like `line == "stop"`. Furthermore, the batch input is editable at runtime, meaning the StopIteration by default would prevent the user from giving the expected input in the same run if the missing line is noticed. Finally, the runtime environment does not know of the existence of any input modes (as we opted to not provide it at all at the start, because it is editable). 
In conclusion, I'm not convinced that the benefits outweigh having to deviate from the standard and how input is handled in Dodona.
